commcare_connect/opportunity/models.py

```diff
@@ -370,11 +425,41 @@ def images(self):
 
 class OpportunityClaim(models.Model):
     opportunity_access = models.OneToOneField(OpportunityAccess, on_delete=models.CASCADE)
-    max_payments = models.IntegerField()
+    # to be removed
+    max_payments = models.IntegerField(null=True)
     end_date = models.DateField()
     date_claimed = models.DateField(auto_now_add=True)
 
 
+class OpportunityClaimLimit(models.Model):
+    opportunity_claim = models.ForeignKey(OpportunityClaim, on_delete=models.CASCADE)
+    payment_unit = models.ForeignKey(PaymentUnit, on_delete=models.CASCADE)
+    max_visits = models.IntegerField()
+
+    @classmethod
+    def create_claim_limits(cls, opportunity: Opportunity, claim: OpportunityClaim):
```

Comment:
Is there a high level description of what this method does?

Reply:
This is basically the same as previous `claim.max_payments` ( I have renamed max_payments to max_visits since visit is what max_payments was also referring to), but because now the max_payments is per `paymentunit` and not just per user/opportunity, we need a per-paymentunit claiming. So, this method does exactly that. It creates per-paymentunit max visit limits